source/server-selection/server-selection.rst

```diff
@@ -1431,6 +1475,39 @@ servers to come and go.  Making newly available servers either first or last
 could lead to unbalanced work.  Random selection has a better fairness
 guarantee and keeps the design simpler.
 
+operationCount-based selection within the latency window (multi-threaded or async)
+----------------------------------------------------------------------------------
+
+When a node begins slows down, any pooled connections opened against it which
```

Comment:
This is a good explanation, this algorithm should handle the case where a node's performance degrades better than random selection. In fact, this algorithm handles even more cases than described here.  For example,  "node begins slows down" could be generalized to "operation execution slows down". "Operation execution" includes factors other than the remote server's performance, like the latency of creating connections and the artificial latency incurred by maxConnecting.

Reply:
Updated it to be a bit more general and use some of the examples you provided.